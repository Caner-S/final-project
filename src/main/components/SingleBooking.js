import React from 'react';
import Paper from '@material-ui/core/Paper';
import {
    makeBooking
} from "../dao/BookingDao";
import firebase from "../../config/firebase";
import Confirm from "./Confirm";
import ConfirmInput from "./ConfirmInput";
import Typography from "@material-ui/core/Typography";
import './SingleBooking.css';


class SingleBooking extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
        };

    }

    businessCase(space){
        if(this.props.businessCase){
            console.log(this.props.arrivalDate);
            return <ConfirmInput buttonText="Request Now" title={"Request a space"} description={"Are you sure you want to make a request for space "+space.data().number+" at "+ new Date(this.props.arrivalDate).toLocaleString() +"?"} spaceId={space.id} user={firebase.auth().currentUser.uid} arrive={this.props.arrivalDate} depart={this.props.departureDate}/>
        } else {
            return <Confirm buttonText="Book Now" title={"Book a space"} description={"Are you sure you want to make a booking for space "+space.data().number+" at "+ new Date(this.props.arrivalDate).toLocaleString() +"?"} onAccept={() => { makeBooking(space.id, firebase.auth().currentUser.uid, this.props.arrivalDate, this.props.departureDate, space.data().number)}} />

        }
    }


    render() {
        let {space} = this.props;
        if(space === undefined){
            return (
            <Paper className="singleBookingPaper">
                <Typography variant="h3">Please select a space on the left</Typography>
            </Paper>);

        }else {

        return (
            <Paper className="singleBookingPaper">
               <Typography variant="h3">Space: {space.data().number}</Typography>
                <Typography variant="subtitle1">{new Date(this.props.arrivalDate).toLocaleString()} to {new Date(this.props.departureDate).toLocaleString()}</Typography>
                <Typography variant="subtitle1">Type: {space.data().type}</Typography>
                <Typography variant="subtitle1">Capacity: {space.data().capacity}</Typography>
                {this.businessCase(space)}
            </Paper>




        );}
    }
}
export default SingleBooking;