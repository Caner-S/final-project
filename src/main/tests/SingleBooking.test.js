import React from 'react';
import renderer from 'react-test-renderer';

import SingleBooking from "../components/SingleBooking";

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<SingleBooking />).toJSON();
    expect(tree).toMatchSnapshot();
});