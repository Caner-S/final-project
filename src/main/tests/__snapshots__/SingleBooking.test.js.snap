// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="MuiPaper-root singleBookingPaper MuiPaper-elevation1 MuiPaper-rounded"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    Please select a space on the left
  </h3>
</div>
`;
